src/api_client.rs

```diff
@@ -89,297 +95,301 @@ impl Client {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_attestations_from_pool(
+    //     slot: Option<Slot>,
+    //     committee_index: Option<CommitteeIndex>,
+    // ) -> Result<Vec<Attestation>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn post_attestations(attestations: &[Attestation]) -> Result<(), Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_attester_slashings_from_pool() -> Result<Vec<AttesterSlashing>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn post_attester_slashing(attester_slashing: &AttesterSlashing) -> Result<(), Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_proposer_slashings_from_pool() -> Result<Vec<ProposerSlashing>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn post_proposer_slashing(proposer_slashing: &ProposerSlashing) -> Result<(), Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn post_sync_committee_messages(
+    //     messages: &[SyncCommitteeMessage],
+    // ) -> Result<(), Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_voluntary_exits_from_pool() -> Result<Vec<SignedVoluntaryExit>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn post_signed_voluntary_exit(exit: &SignedVoluntaryExit) -> Result<(), Error> {
+    //     unimplemented!("")
+    // }
+
+    // /* config namespace */
+    // pub async fn get_fork_schedule() -> Result<Vec<Fork>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_spec() -> Result<HashMap<String, String>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_deposit_contract_address() -> Result<(ChainId, ExecutionAddress), Error> {
+    //     unimplemented!("")
+    // }
+
+    // /* debug namespace */
+    // // v2 endpoint
+    // pub async fn get_state(id: StateId) -> Result<BeaconState, Error> {
+    //     unimplemented!("")
+    // }
+
+    // // v2 endpoint
+    // pub async fn get_heads() -> Result<Vec<Coordinate>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // /* events namespace */
+    // // TODO: figure out return type
+    // pub async fn get_events<T>(topics: &[EventTopic]) -> Result<T, Error> {
+    //     // get back "event: TOPIC, data: T"
+    //     unimplemented!("")
+    // }
+
+    // /* node namespace */
+    // pub async fn get_node_identity() -> Result<NetworkIdentity, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_node_peers(filters: &[PeerDescriptor]) -> Result<Vec<PeerDescription>, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_peer(peer_id: Multiaddr) -> Result<PeerDescription, Error> {
+    //     unimplemented!("")
+    // }
+
+    // pub async fn get_peer_count() -> Result<PeerSummary, Error> {
+    //     unimplemented!("")
+    // }
+
+    pub async fn get_node_version(self) -> ApiResult<String> {
```

Comment:
going to want to take self by reference, otherwise you would consume the `Client` on this call and only be able to make one call with it
```suggestion
    pub async fn get_node_version(&self) -> ApiResult<String> {
```